perl/lib/NeedRestart/UI/Debconf.pm

```diff
@@ -158,31 +158,35 @@ sub command {
 }
 
 
-sub query_pkgs($$$$$$) {
+sub query_pkgs($$$$$$$) {
     my $self = shift;
     my $out = shift;
     my $defno = shift;
     my $pkgs = shift;
     my $overrides = shift;
     my $cb = shift;
+    my $exitWhenNoneSelected = shift;
 
     # prepare checklist array
     my @l = nsort keys %$pkgs;
 
     # apply rc selection overrides
     my @selected = ();
     foreach my $pkg (@l) {
-	my $found;
-	foreach my $re (keys %$overrides) {
-	    next unless($pkg =~ /$re/);
+		my $found;
+		foreach my $re (keys %$overrides) {
+		    next unless($pkg =~ /$re/);
+	
+		    push(@selected, $pkg) if($overrides->{$re});
+		    $found++;
+		    last;
+		}
+	
+		push(@selected, $pkg) unless($defno || $found);
+    }
```

Comment:
Can you please keep the indention? I'm aware that it is mixed up with spaces and tabs all over in the sources of needrestart :-(. Changing it here makes it harder for me to understand the changes.